examples/molnet/train_molnet.py

```diff
@@ -0,0 +1,205 @@
+    else:
+        print('preprocessing dataset...')
+        preprocessor = preprocess_method_dict[method]()
+        # only use first 100 for debug if num_data >= 0
+        target_index = numpy.arangs(num_data) if num_data >= 0 else None
+        datasets = D.molnet.get_molnet_dataset(dataset_name, preprocessor,
+                                               labels=labels,
+                                               target_index=target_index)
+        if not os.path.exists(cache_dir):
+            os.mkdir(cache_dir)
+        datasets = datasets['dataset']
+        for i, fpath in enumerate(filepaths):
+            NumpyTupleDataset.save(fpath, datasets[i])
+
+    train, val, _ = datasets
+
+    # Network
+    if method == 'nfp':
+        print('Train NFP model...')
+        model = GraphConvPredictor(NFP(out_dim=n_unit, hidden_dim=n_unit,
+                                       n_layers=conv_layers),
+                                   MLP(out_dim=class_num, hidden_dim=n_unit))
+    elif method == 'ggnn':
+        print('Train GGNN model...')
+        model = GraphConvPredictor(GGNN(out_dim=n_unit, hidden_dim=n_unit,
+                                        n_layers=conv_layers),
+                                   MLP(out_dim=class_num, hidden_dim=n_unit))
+    elif method == 'schnet':
+        print('Train SchNet model...')
+        model = GraphConvPredictor(
+            SchNet(out_dim=class_num, hidden_dim=n_unit, n_layers=conv_layers),
+            None)
+    elif method == 'weavenet':
+        print('Train WeaveNet model...')
+        n_atom = 20
+        n_sub_layer = 1
+        weave_channels = [50] * conv_layers
+        model = GraphConvPredictor(
+            WeaveNet(weave_channels=weave_channels, hidden_dim=n_unit,
+                     n_sub_layer=n_sub_layer, n_atom=n_atom),
+            MLP(out_dim=class_num, hidden_dim=n_unit))
+    elif method == 'rsgcn':
+        print('Train RSGCN model...')
+        model = GraphConvPredictor(
+            RSGCN(out_dim=n_unit, hidden_dim=n_unit, n_layers=conv_layers),
+            MLP(out_dim=class_num, hidden_dim=n_unit))
+    else:
+        raise ValueError('[ERROR] Invalid method {}'.format(method))
+
+    train_iter = iterators.SerialIterator(train, args.batchsize)
+    val_iter = iterators.SerialIterator(val, args.batchsize,
+                                        repeat=False, shuffle=False)
+
+    metrics_fun = molnet_default_config[dataset_name]['metrics']
+    loss_fun = molnet_default_config[dataset_name]['loss']
+    if molnet_default_config[dataset_name]['task_type'] == 'regression':
+        model = Regressor(model, lossfun=loss_fun, metrics_fun=metrics_fun,
+                          device=args.gpu)
+    elif molnet_default_config[args.dataset]['task_type'] == 'classification':
+        model = Classifier(model, lossfun=loss_fun, metrics_fun=metrics_fun,
+                           device=args.gpu)
+    # TODO(motoki): how to support bace dataset
```

Comment:
we can split "dataset_type" itself?

Reply:
Sorry this comment is confusing. At that time `dataset_type` of bace dataset is 'mix' (classification and regression).

Reply:
I read the MoleculeNet paper, and they are supporting MultiTask network training. It was written that sharing the network weight can reduce validation error. So we should try classification + regression multitask formulation.